### PR TITLE
Hubspot Integration

### PR DESCRIPTION
**Pull request by Lincoln23**
Currently able to pull data from the Company's API and Contact's API

### CODE REVIEW

File: integrations/manual/hubspot/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+## Hubspot Integraiton
+
+
+``` 
+npm install hubspot
```
Comment (bobby-brennan) [inline integrations/manual/hubspot/README.md]: This should be `@datafire/hubspot`

File: integrations/manual/hubspot/index.js [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+    description: "your company id",
+    type: "Integer",
+  }, {
+    name: "vidOffset",
+    description: "The vid to offset the query by.",
+    type: 'Integer',
+    default: ""
+  }, {
+    name: "count",
+    description: "Specifies the number of contacts to be returned. Supports a maximum value of 600.",
```
Comment (bobby-brennan) [inline integrations/manual/hubspot/index.js]: Not necessary, but as an FYI you can set `maximum: 600` here.

File: integrations/manual/hubspot/index.js [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+    default: ""
+  }, {
+    name: "property_mode",
+    description: "One of “value_only” or “value_and_history” to specify if the current value for a property should be fetched, or the value and all the historical values for that property. Default is “value_only”",
+    default: "value_only",
+    type: "string",
+  }, {
+    name: "form_submission_mode",
+    description: "One of “all”, “none”, “newest”, “oldest” to specify which form submissions should be fetched. Default is “newest",
+    default: "newest",
```
Comment (bobby-brennan) [inline integrations/manual/hubspot/index.js]: Also as an FYI - you can set `enum: ['all', 'none', 'newest', 'oldest']`

File: integrations/manual/hubspot/package.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "hubspot_integration",
```
Comment (bobby-brennan) [inline integrations/manual/hubspot/package.json]: This should be `@datafire/hubspot`

**Comment by Lincoln23** [comment]
Thanks! I applied those changes

**Comment by bobby-brennan** [comment]
Thanks Lincoln. Looks like there were a few issues with the code here - I've fixed some of the biggest ones (you can see my [changes here](https://github.com/DataFire/Integrations/commits/master/integrations/manual/hubspot/index.js))

Do you mind trying out each of the actions you implemented to make sure they work before I publish this?

**Comment by Lincoln23** [comment]
I am trying to test these actions right now, however, I am getting this error
`error: "Account hubspot not specified for action hubspot. Did you remember to pass in the 'context' object?"`

I am putting my credentials in `hubspot.security.fields` How do I create a `context` object?

**Comment by bobby-brennan** [comment]
The easiest way to test the actions is to run:

```bash
datafire authenticate hubspot
```

This will ask you for a client_id and client_secret, which you can get at https://app.hubspot.com. Make sure to select the same scopes in the command line that you added for your app on Hubspot.com

Then follow the link output by the `datafire authenticate` command. This will create DataFire-accounts.yml, which will contain your credentials. Then you can run e.g.:

```bash
datafire run hubspot/getAllCustomers
```

Hope that helps - let me know if you have trouble

**Comment by Lincoln23** [comment]
To test all the actions I added hubspot to  `node modules` are ran `datafire authenticate` and `datafire run hubspot/action` to test
screenshot_1

 I made some fixes and I test every action. Everything is working. 

There is one small thing I wasn't sure how to fix though. In the `inputs array`, when I define the type as `integer`  I am not sure how to set the default value as `null` (putting `default: null` didn't seem to work). 
so when I was testing I just changed these to type `string` and default as `""`. 
I was wondering if you could fix that
This is on 
line `#140`  for vidOffset, 
on line `#166` for vidoffset, 
on line `#327` for time_offset and 
 on line `#332` for contact_offset

Other than that everything works perfectly!

**Comment by bobby-brennan** [comment]
Thanks Lincoln!

For `integer` types, the default needs to be an integer. You can just leave the default field blank if there is no default.

**Comment by Lincoln23** [comment]
No problem, should I start a new pull request for my latest version?

**Comment by bobby-brennan** [comment]
Hey Lincoln, sorry for the delay, this got lost in my inbox.

Yes, if you could open a new pull request that would be great.

**Comment by Lincoln23** [comment]
ok I have submitted a new request

**Pull request merged**